File: 백준/Silver/1654. 랜선 자르기/랜선 자르기.java
import java.io.*;
import java.util.StringTokenizer;

class Main{
    public static void main(String args[])throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int k = Integer.parseInt(st.nextToken());
        int n = Integer.parseInt(st.nextToken());
        long max = 0;
        int[] arr = new int[k];
        for(int i=0;i<k;i++){
            arr[i] = Integer.parseInt(br.readLine());
            if(max<arr[i])
                max = arr[i];
        }
        max++;
        long min=0;
        long mid=0;
        while(min < max){
            mid = (max+min)/2;
            long count=0;
            for(int i=0;i<arr.length;i++)
                count+=(arr[i]/mid);
            if(count<n)
                max=mid;
            else
                min = mid+1;
        }
        bw.write(min-1+"\n");
        bw.close();
    }

}

File: 백준/Silver/1654. 랜선 자르기/랜선 자르기.cc
#include <iostream>

#include<vector>

using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(nullptr);
	long min=0, mid = 0,max = 0;
	int n, k;
	cin >> n >> k;
	vector<int> vec = vector<int>(n, 0);
	for(int i=0;i<n;i++) {
		cin >> vec[i];
		if (max < vec[i])
			max = vec[i];
	}
	max++;
	while (min < max) {
		mid = (max + min) / 2;
		int count = 0;
		for (int i = 0; i < n; i++) {
			count += vec[i] / mid;
		}
		if (k > count) {
			max = mid;
		}
		else {
			min = mid + 1;
		}
	}
	cout << max - 1;
	return 0;
}